### ISSUE TITLE
Warning spam in Compatibility renderer: Attempting to use a shader that requires tangents with a mesh that doesn't contain tangents. 

### ISSUE THREAD

**Issue by jsjtxietian** [godotengine/godot#84418]
### Godot version

Godot v4.2.beta (5ee983188)

### System information

 - Windows 10.0.22621 - GLES3 (Compatibility) - NVIDIA GeForce RTX 3060 Ti (NVIDIA; 31.0.15.3699) - 12th Gen Intel(R) Core(TM) i7-12700F (20 Threads)

### Issue description

When create a new scene and add any mesh instance 3d to it, the following warning spams: 
```
drivers\gles3\rasterizer_scene_gles3.cpp:265 - Attempting to use a shader that requires tangents with a mesh that doesn't contain tangents. Ensure that meshes are imported with the 'ensure_tangents' option. If creating your own meshes, add an `ARRAY_TANGENT` array (when using ArrayMesh) or call `generate_tangents()` (when using SurfaceTool). (User)
```

This doen't happen in forward plus or mobile.

![image](https://github.com/godotengine/godot/assets/8315986/90e7b434-9ccc-4682-a0a5-cd4f417a5db0)

### Steps to reproduce

1. Create a new project with Compatibility renderer
2. create a new scene and add a mesh instance 3d to it
3. observe the warning spam

### Minimal reproduction project

N/A

**Comment by chrisl8** [comment]
Just start an empty project in Compatibility mode in 4.2 Beta and this happens immediately without adding any meshes.
<img width="1918" alt="image" src="https://github.com/godotengine/godot/assets/6188278/b207c0fd-ac0c-4995-b720-4dc1f9b13c02">

**Comment by chrisl8** [comment]
This warning appears to have been introduced by this commit:

https://github.com/godotengine/godot/commit/d1043a5f930513aa4b48b014ec7961cad48b0560

**Comment by clayjohn** [comment]
We uses the `uses_tangent` variable to track whether the shader uses tangents or not. The variable is set to `true` if certain shader built ins are used here: https://github.com/godotengine/godot/blob/3e7f638d7b574785f521beafaf52a6ad95be016f/drivers/gles3/storage/material_storage.cpp#L2925-L2928

However, mistakenly `uses_tangent` isn't initialized above (and a few others). So if it's not set to `true` it's used uninitialized and can be anything. 

I'll bet that is what's causing this issue and the solution is to initialize it to false above with all the other variables.

**Comment by jsjtxietian** [comment]
I will take a look, thanks!

**Comment by akien-mga** [comment]
Confirmed with 4.2-beta5 official build. Can't reproduce with a local dev build of the master branch, probably because GCC initializes memory differently for unoptimized builds.

**Comment by chrisl8** [comment]
I'm definitely seeing it on my local build from master still, but I'm building on Windows so I'm not using gcc I guess.

**Comment by WalruswareGames** [comment]
This is marked as closed but in godot 4.2.beta6 the issue is still there. This screenshot is from a completely new project, no nodes or scripts at all. Windows 11. 

![Screenshot 2023-11-15 195504](https://github.com/godotengine/godot/assets/94482762/77421484-2130-42e4-8507-52f355c71ffa)

**Comment by chrisl8** [comment]
@WalruswareGames This didn't make it into Beta 6, but it is in the master branch, so it will be in the next beta/rc release.

**Comment by daveTheOldCoder** [comment]
If it's not in beta6, why did it resolve this issue for me? 

Godot v4.2.beta6 - Pop!_OS 20.04 LTS - X11 - GLES3 (Compatibility) - Mesa Intel(R) UHD Graphics 630 (CFL GT2) () - Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz (16 Threads)

**Comment by clayjohn** [comment]
> > This didn't make it into Beta 6
> 
> If it's not in beta6, why did upgrading from beta5 to beta6 resolve this issue for me?
> 
> Godot v4.2.beta6 - Pop!_OS 20.04 LTS - X11 - GLES3 (Compatibility) - Mesa Intel(R) UHD Graphics 630 (CFL GT2) () - Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz (16 Threads)

It's an uninitialized memory error. So it happens randomly. It's only working for you in beta 6 by chance. 

The actual fix is coming in the next release

**Comment by WalruswareGames** [comment]
Thanks for clarifying and sorry for the bother. I didn't want to not
mention it in case it was still broken I didn't realise it hadn't been
included in beta 6.

**Comment by salt0fLife** [comment]
just started happening randomly to me. This project is months old and has countless imported meshes, and I only just started having an issue. 
I am running godot steam 4.2
It started right after I duplicated a "made local" imported mesh. not sure if that had anything to do with it, but that could be some sort of trigger.
I deleted all versions of the duplicated mesh and the warning spam persists. I am now using an identical version of my project that previously did not have this issue. 

I can live with this for now, but I hope a solution is found soon
